### PR TITLE
Significant Picross Improvements

### PR DESCRIPTION
**Pull request by hunterdyar**
I think the Picross game currently in the main branch crashes. Still hard at work on it, but mostly on level design and small visual fixes. It's a good time to get more feedback.

(Note: changing read/write perma from int32 to blob is still todo, but planned)

### CODE REVIEW

File: main/modes/picross/mode_picross.c
Comment (AEFeinstein) [inline main/modes/picross/mode_picross.c:1015]: There's a function called `textWidth()` which you give a font and a string and it returns the pixel width of the string. It's better than `strlen()` and a guess at the per-char width.

File: main/modes/picross/mode_picross.c
Comment (AEFeinstein) [inline main/modes/picross/mode_picross.c:1371]: `main` does have functions now to read and write blobs of binary data in NVM. If you wanted to, you could make a struct with all the picross data and do a single read/write, rather than banking it into 32 bit ints. Probably cleaner, also a bit more work from where you are. Your call.
```C
bool readNvsBlob(const char* key, void* out_value, size_t* length);
bool writeNvsBlob(const char* key, const void* value, size_t length);
```

File: main/modes/picross/mode_picross.c
Comment (AEFeinstein) [inline main/modes/picross/mode_picross.c:1459]: it could be `sprintf(bankName, "pic_b_%d", i);` (or maybe banks aren't necessary at all with blobs)

File: sdkconfig
Comment (AEFeinstein) [inline sdkconfig:53]: Not quite sure why this got changed, but can you revert it?

File: main/modes/picross/picross_menu.c
Comment (AEFeinstein) [inline main/modes/picross/picross_menu.c:86]: i think you can have it be `.modeName = str_picrossTitle,` in order to guarantee a match

**Review by AEFeinstein — CHANGES_REQUESTED** [review]
The only thing I need you to fix before merging is reverting the `sdkconfig` file. You should probably fix the text centering too (easy), and maybe blobify save data (less easy)

File: sdkconfig
Comment (hunterdyar) [reply sdkconfig:53]: Resolved just the changes that were made to sdkconfig from that commit, everything still runs fine on my emulator-only end, i'm sure we will want to double check things.

File: main/modes/picross/mode_picross.c
Comment (hunterdyar) [reply main/modes/picross/mode_picross.c:1015]: Thanks! Easy fix with, now knowing that function exists.

File: sdkconfig
Comment (AEFeinstein) [reply sdkconfig:53]: Ayep, definitely double checking over here

File: main/modes/picross/mode_picross.c
Comment (AEFeinstein) [reply main/modes/picross/mode_picross.c:1015]: Knowing is half the battle!

File: main/modes/picross/mode_picross.c
Comment (hunterdyar) [reply main/modes/picross/mode_picross.c:1361]: oof, but done.

Humerously, the JSON file in the emulator serializes it way LESS efficiently now. An array of bools is now serialized as a string of 1s and 0s with each bool taking 2 spaces. Similar silliness with the 2d array of enums that is the level. I believe this is the emu JSON serializer, but "fyi" because perhaps I am just doing something wrong.

File: main/modes/picross/picross_select.c
Comment (hunterdyar) [reply main/modes/picross/picross_select.c:239]: I'll circle back on this later this week when I have more time. I plan to; as I would like the headroom to explore the possibility of a more sophisticated hint system.

**Comment by hunterdyar** [comment]
> The only thing I need you to fix before merging is reverting the `sdkconfig` file. You should probably fix the text centering too (easy), and maybe blobify save data (less easy)

sdkconfig fixed, text centering fixed, blob save data also done. drawwsg improvements left planned for future-me.

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**
